### ISSUE TITLE
[BUG] configuration fails if googletest doesn't provide gmock

### ISSUE THREAD

**Issue by upsj** [rapidsai/rapids-cmake#237]
**Describe the bug**
CPM picks up a locally installed googletest library regardless of whether it provides `GTest::gmock` (whose presence can be controlled via `-DBUILD_GMOCK=OFF`). This leads to the following error on generation:
```
-- CPM: using local package GTest@1.10.0
-- CPM: adding package KvikIO@22.08 (branch-22.08)
-- Found cuFile: /usr/local/cuda-11.7/lib64/libcufile.so
-- Configuring done
CMake Error at CMakeLists.txt:692 (target_link_libraries):
  Target "cudftestutil" links to:

    GTest::gmock

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

**Steps/Code to reproduce bug**
Install googletest 1.8.0 or newer from source with `-DBUILD_GMOCK=OFF` and run `build.sh`.
I used [spack](https://github.com/spack/spack) via `spack install googletest` to install the library.

**Expected behavior**
The cmake configuration should work

**Environment overview (please complete the following information)**
 - Environment location: WSL2 with Ubuntu 20.04
 - Method of cuDF install: from source

**Environment details**
<details><summary>Click here to see environment details</summary><pre>

     **git***
     commit e2ff00f665472a477f7c5b90bed8045c9d0d40a4 (HEAD -> branch-22.08, origin/branch-22.08, origin/HEAD)
     Author: etseidl <etseidl@users.noreply.github.com>
     Date:   Mon Jul 4 14:32:08 2022 -0700

     Add thrift support for parquet column and offset indexes (#11178)

     Adds some necessary structs to parquet.hpp as well as methods to CompactProtocolReader/Writer to address rapidsai/cudf#9268

     I can add tests if necessary once rapidsai/cudf#11177 is merged, or testing can be deferred to be included in a future PR (based on rapidsai/cudf#11171)

     Authors:
     - https://github.com/etseidl

     Approvers:
     - Devavret Makkar (https://github.com/devavret)
     - Yunsong Wang (https://github.com/PointKernel)

     URL: https://github.com/rapidsai/cudf/pull/11178
     **git submodules***

     ***OS Information***
     DISTRIB_ID=Ubuntu
     DISTRIB_RELEASE=20.04
     DISTRIB_CODENAME=focal
     DISTRIB_DESCRIPTION="Ubuntu 20.04 LTS"
     NAME="Ubuntu"
     VERSION="20.04 LTS (Focal Fossa)"
     ID=ubuntu
     ID_LIKE=debian
     PRETTY_NAME="Ubuntu 20.04 LTS"
     VERSION_ID="20.04"
     HOME_URL="https://www.ubuntu.com/"
     SUPPORT_URL="https://help.ubuntu.com/"
     BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
     PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
     VERSION_CODENAME=focal
     UBUNTU_CODENAME=focal
     Linux 5.10.16.3-microsoft-standard-WSL2 rapidsai/cudf#1 SMP Fri Apr 2 22:23:49 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

     ***GPU Information***
     Tue Jul  5 15:54:11 2022
     +-----------------------------------------------------------------------------+
     | NVIDIA-SMI 515.43.01    Driver Version: 516.01       CUDA Version: 11.7     |
     |-------------------------------+----------------------+----------------------+
     | GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
     | Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
     |                               |                      |               MIG M. |
     |===============================+======================+======================|
     |   0  Quadro RTX 3000...  On   | 00000000:01:00.0 Off |                  N/A |
     | N/A   48C    P0    18W /  N/A |      0MiB /  6144MiB |      0%      Default |
     |                               |                      |                  N/A |
     +-------------------------------+----------------------+----------------------+

     +-----------------------------------------------------------------------------+
     | Processes:                                                                  |
     |  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
     |        ID   ID                                                   Usage      |
     |=============================================================================|
     |  No running processes found                                                 |
     +-----------------------------------------------------------------------------+

     ***CPU***
     Architecture:                    x86_64
     CPU op-mode(s):                  32-bit, 64-bit
     Byte Order:                      Little Endian
     Address sizes:                   39 bits physical, 48 bits virtual
     CPU(s):                          16
     On-line CPU(s) list:             0-15
     Thread(s) per core:              2
     Core(s) per socket:              8
     Socket(s):                       1
     Vendor ID:                       GenuineIntel
     CPU family:                      6
     Model:                           165
     Model name:                      Intel(R) Core(TM) i7-10875H CPU @ 2.30GHz
     Stepping:                        2
     CPU MHz:                         2304.007
     BogoMIPS:                        4608.01
     Hypervisor vendor:               Microsoft
     Virtualization type:             full
     L1d cache:                       256 KiB
     L1i cache:                       256 KiB
     L2 cache:                        2 MiB
     L3 cache:                        16 MiB
     Vulnerability Itlb multihit:     KVM: Mitigation: VMX unsupported
     Vulnerability L1tf:              Not affected
     Vulnerability Mds:               Not affected
     Vulnerability Meltdown:          Not affected
     Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl and seccomp
     Vulnerability Spectre v1:        Mitigation; usercopy/swapgs barriers and __user pointer sanitization
     Vulnerability Spectre v2:        Mitigation; Enhanced IBRS, IBPB conditional, RSB filling
     Vulnerability Srbds:             Not affected
     Vulnerability Tsx async abort:   Not affected
     Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology cpuid pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt xsaveopt xsavec xgetbv1 xsaves flush_l1d arch_capabilities

     ***CMake***
     /home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/cmake-3.23.2-p2knxqhaxheekzcizgr25xxj2vzahssk/bin/cmake
     cmake version 3.23.2

     CMake suite maintained and supported by Kitware (kitware.com/cmake).

     ***g++***
     /home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gcc-11.3.0-smnfbmjldfjl6utm5wuabw5sntrgag5z/bin/g++
     g++ (Spack GCC) 11.3.0
     Copyright (C) 2021 Free Software Foundation, Inc.
     This is free software; see the source for copying conditions.  There is NO
     warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

     ***nvcc***
     /usr/local/cuda-11.7/bin/nvcc
     nvcc: NVIDIA (R) Cuda compiler driver
     Copyright (c) 2005-2022 NVIDIA Corporation
     Built on Tue_May__3_18:49:52_PDT_2022
     Cuda compilation tools, release 11.7, V11.7.64
     Build cuda_11.7.r11.7/compiler.31294372_0

     ***Python***
     /home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/python-3.9.13-mjbtlo67lldhazprooi7bscvnenehw56/bin/python
     Python 3.9.13

     ***Environment Variables***
     PATH                            : /home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/openssl-1.1.1p-suhytzhenyctx4hskzud246ul7mh7mse/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gcc-11.3.0-smnfbmjldfjl6utm5wuabw5sntrgag5z/bin:/usr/local/cuda-11.7/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/hwloc-2.7.1-v2ajufvqih6ahfobdfqut5brvgwxle3p/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gflags-2.2.2-sfpwril4k2r5cwatrmepfdvxjl542tkf/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/openmpi-4.1.4-57ki2uxf2tivfambnt7wmt76synav4on/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/python-3.9.13-mjbtlo67lldhazprooi7bscvnenehw56/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/ninja-1.11.0-ihlo3n7cyhvrh6oxalsks25edlcq3g3j/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/cmake-3.23.2-p2knxqhaxheekzcizgr25xxj2vzahssk/bin:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-14.0.6-gkek3xptdok6lahz7lqj3eyq3xi7xjdz/bin:/home/tribizel/spack/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib:/mnt/c/Program
     Files/PowerShell/7:/mnt/c/Program: Files/NVIDIA
     GPU                             : Computing
     Toolkit/CUDA/v11.7/bin:/mnt/c/Program: Files/NVIDIA
     GPU                             : Computing
     Toolkit/CUDA/v11.7/libnvvp:/mnt/c/windows/system32:/mnt/c/windows:/mnt/c/windows/System32/Wbem:/mnt/c/windows/System32/WindowsPowerShell/v1.0:/mnt/c/windows/System32/OpenSSH:/mnt/c/Program: Files/Perforce:/mnt/c/Program
     Files/NVIDIA                    : Corporation/Nsight
     Compute                         : 2022.2.0:/mnt/c/Program
     Files                           : (x86)/NVIDIA
     Corporation/PhysX/Common:/mnt/c/Users/tribizel/AppData/Local/Microsoft/WindowsApps:/mnt/c/Users/tribizel/AppData/Local/Programs/Microsoft: VS
     Code/bin:/snap/bin              :
     LD_LIBRARY_PATH                 : /home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/googletest-1.10.0-52m35fglkskemxojpgknbcebrnlpafx6/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/openssl-1.1.1p-suhytzhenyctx4hskzud246ul7mh7mse/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/zlib-1.2.12-3xrlqm2oj6djouizyxcwt7d36zifpibd/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gcc-11.3.0-smnfbmjldfjl6utm5wuabw5sntrgag5z/lib64:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gcc-11.3.0-smnfbmjldfjl6utm5wuabw5sntrgag5z/lib:/usr/local/cuda-11.7/lib64:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/rapidjson-1.2.0-2022-03-09-cwptm7yij2im2dvbd2o4teywroik3lqq/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/benchmark-1.6.0-a5yaa2qhx45pm3s4i3nyeosdiir3ffuu/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/hwloc-2.7.1-v2ajufvqih6ahfobdfqut5brvgwxle3p/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/gflags-2.2.2-sfpwril4k2r5cwatrmepfdvxjl542tkf/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/openmpi-4.1.4-57ki2uxf2tivfambnt7wmt76synav4on/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/python-3.9.13-mjbtlo67lldhazprooi7bscvnenehw56/lib:/home/tribizel/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.4.0/llvm-14.0.6-gkek3xptdok6lahz7lqj3eyq3xi7xjdz/lib
     NUMBAPRO_NVVM                   :
     NUMBAPRO_LIBDEVICE              :
     CONDA_PREFIX                    :
     PYTHON_PATH                     :

     conda not found
     pip not found
</pre></details>

**Additional context**

**Comment by robertmaynard** [comment]
Looking into this more we can't easily work-around the problem.

If we allow `find_package(GTest)` to execute it will construct the `GTest::gtest` and `GTest::gtest_main` targets, therefore we can't fall back to building from source as the target names already exist ( you can't delete targets in CMake ).

The best we can do is provide better error message to the user with remidation recommendations.
